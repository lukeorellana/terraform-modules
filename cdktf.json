{
  "language": "typescript",
  "app": "npx ts-node",
  "terraformProviders": [],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb"
}