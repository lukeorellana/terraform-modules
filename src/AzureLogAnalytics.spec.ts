import { Testing, TerraformStack} from 'cdktf';
import {AzurermProvider} from "@cdktf/provider-azurerm";
import { AzureLogAnalytics } from './AzureLogAnalytics';
import 'cdktf/lib/testing/adapters/jest';


// test('renders a VPC with minimal config', () => {
//   const synthed = Testing.synthScope((stack) => {
//     new PocketVPC(stack, 'testPocketVPC');
//   });


describe('AzureLogAnalytics-Snapshot', () => {
  it('renders a AzureLogAnalytics without tags', () => {

    const synthed = Testing.synthScope((stack) => {
      new AzureLogAnalytics(stack, 'testLA', {
        name: 'la-test',
        location: 'eastus',
        retention: 90,
        sku: "PerGB2018",
        resource_group_name: "rg-test",
      });
    });
  
    expect(synthed).toMatchSnapshot();
  });

  it('renders a AzureLogAnalytics with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new AzureLogAnalytics(stack, 'testLA', {
        name: 'la-test',
        location: 'eastus',
        tags: {
          environment: "test",
        },
        retention: 90,
        sku: "PerGB2018",
        resource_group_name: "rg-test",
      });
    });
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureLogAnalytics-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });


    new AzureLogAnalytics(stack, 'testLA', {
      name: 'la-test',
      location: 'eastus',
      tags: {
        environment: "test",
      },
      retention: 90,
      sku: "PerGB2018",
      resource_group_name: "rg-test",
    });


    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });

    new AzureLogAnalytics(stack, 'testLA', {
      name: 'la-test',
      location: 'eastus',
      tags: {
        environment: "test",
      },
      retention: 90,
      sku: "PerGB2018",
      resource_group_name: "rg-test",
    });



    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
