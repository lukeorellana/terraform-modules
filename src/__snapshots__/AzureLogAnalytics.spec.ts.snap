// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureLogAnalytics-Snapshot renders a AzureLogAnalytics with tags 1`] = `
"{
  "output": {
    "log_analytics_id": {
      "value": "\${azurerm_log_analytics_workspace.testLA_log_analytics_D32C1799.id}"
    },
    "log_analytics_primary_shared_key": {
      "sensitive": true,
      "value": "\${azurerm_log_analytics_workspace.testLA_log_analytics_D32C1799.primary_shared_key}"
    },
    "log_analytics_resource_group": {
      "value": "rg-test"
    },
    "log_analytics_workspace_id": {
      "value": "\${azurerm_log_analytics_workspace.testLA_log_analytics_D32C1799.workspace_id}"
    }
  },
  "resource": {
    "azurerm_log_analytics_workspace": {
      "testLA_log_analytics_D32C1799": {
        "location": "eastus",
        "name": "la-test",
        "resource_group_name": "rg-test",
        "retention_in_days": 90,
        "sku": "PerGB2018",
        "tags": {
          "environment": "test"
        }
      }
    }
  }
}"
`;

exports[`AzureLogAnalytics-Snapshot renders a AzureLogAnalytics without tags 1`] = `
"{
  "output": {
    "log_analytics_id": {
      "value": "\${azurerm_log_analytics_workspace.testLA_log_analytics_D32C1799.id}"
    },
    "log_analytics_primary_shared_key": {
      "sensitive": true,
      "value": "\${azurerm_log_analytics_workspace.testLA_log_analytics_D32C1799.primary_shared_key}"
    },
    "log_analytics_resource_group": {
      "value": "rg-test"
    },
    "log_analytics_workspace_id": {
      "value": "\${azurerm_log_analytics_workspace.testLA_log_analytics_D32C1799.workspace_id}"
    }
  },
  "resource": {
    "azurerm_log_analytics_workspace": {
      "testLA_log_analytics_D32C1799": {
        "location": "eastus",
        "name": "la-test",
        "resource_group_name": "rg-test",
        "retention_in_days": 90,
        "sku": "PerGB2018"
      }
    }
  }
}"
`;
